packages/test-runner/src/runner.ts

```diff
@@ -45,15 +45,37 @@ export class Runner {
   private _afterFunctions: Function[] = [];
   private _parameterValues: Matrix = {};
 
-  constructor(config: RunnerConfig, files: string[], reporter: Reporter) {
+  constructor(config: RunnerConfig, reporter: Reporter) {
     this._config = config;
     this._reporter = reporter;
+  }
+
+  parameters(): ParameterRegistration[] {
+    return [...parameterRegistrations.values()];
+  }
+
+  setParameterValue(name: string, value: string) {
+    this.setParameterValues(name, [value]);
+  }
+
+  setParameterValues(name: string, values: string[]) {
+    if (!parameterRegistrations.has(name))
+      throw new Error(`Unregistered parameter '${name}' was set.`);
+    this._parameterValues[name] = values;
+  }
 
+  loadFiles(files: string[]): boolean {
     // First traverse tests.
     for (const file of files) {
       const suite = new Suite('');
-      const revertBabelRequire = spec(suite, file, config.timeout, undefined);
-      require(file);
+      const revertBabelRequire = spec(suite, file, this._config.timeout, undefined);
+      try {
+        require(file);
+      } catch (error) {
+        revertBabelRequire();
+        this._reporter.onLoadingFailure(file, serializeError(error));
```

Comment:
Since we consider files to be independent, would it make sense to not fail everything on an error in a single file?

Reply:
Yes. That matches what jest does. However it gets complicated with reporting and us loading files twice. I decided not to change semantics in this patch.

Reply:
I had a local patch with the additional `onSuiteError` signal, I think it was ok, but I ended up not needing it. Should be fairly easy to add it, fine to do in another patch. File a bug?